pkg/vault/contracts/Vault.sol

```diff
@@ -647,6 +648,10 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             poolData.poolConfigBits.requireUnbalancedLiquidityEnabled();
 
             amountsInScaled18 = maxAmountsInScaled18;
+            // Deep copy given max amounts in raw to calculated amounts in raw to avoid scaling later, ensuring that
+            // `maxAmountsIn` is preserved.
+            ScalingHelpers.copyToArray(params.maxAmountsIn, amountsInRaw);
+
```

Comment:
Why do we need to do a deep copy here? the amountsInRaw values are not edited, unless it's 0 (which would probably not enter this `if` clause anyway).

Reply:
True, we don't strictly need it, but you need to reason about that to figure out that you're not really modifying `maxAmountsIn`. Sounds a bit safer to just do a copy.

Reply:
And I guess this is the only place, since we don't have a remove with exact (multiple) tokens out.

Reply:
Correct.
More so, unless we create the array inside each the block corresponding to each kind, we've already allocated the memory by this point. The copy only loops and assigns the values from the given argument to the new, empty array.